plugins/kubernetes-backend/src/service/KubernetesProxy.ts

```diff
@@ -98,26 +98,29 @@ export class KubernetesProxy {
         req.header('authorization'),
       );
 
-      const authorizeResponse = (
-        await permissionApi.authorize(
-          [{ permission: kubernetesProxyPermission }],
-          {
-            token,
-          },
-        )
-      )[0];
-
-      if (authorizeResponse.result === AuthorizeResult.DENY) {
+      const authorizeResponse = await permissionApi.authorize(
+        [{ permission: kubernetesProxyPermission }],
+        {
+          token,
+        },
+      );
+      const auth = authorizeResponse[0];
+
+      if (auth.result === AuthorizeResult.DENY) {
         res.status(403).json({ error: new NotAllowedError('Unauthorized') });
         return;
       }
 
-      const cluster = await this.getClusterForRequest(req).then(cd =>
-        this.authTranslator.decorateClusterDetailsWithAuth(cd, {}),
-      );
-      if (!req.headers.authorization) {
-        req.headers.authorization = `Bearer ${cluster.serviceAccountToken}`;
-      }
+      req.headers.authorization = req.header(HEADER_KUBERNETES_AUTH)
+        ? req.header(HEADER_KUBERNETES_AUTH)
+        : `Bearer ${
```

Comment:
I think you can use a [nullish coalescing operator](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#nullish-coalescing) here to tighten it even more?
```suggestion
      req.headers.authorization = req.header(HEADER_KUBERNETES_AUTH)
        ?? `Bearer ${
```

Reply:
Done